liquid-compiler/src/parser.rs

```diff
@@ -194,12 +220,27 @@ fn parse_filter(filter: Pair, options: &Language) -> Result<FilterCall> {
             Error::with_msg("Unknown filter")
                 .context("requested filter", name.to_owned())
                 .context("available filters", available)
-        })?
-        .clone();
-    let f = FilterCall::new(name, f, args);
+        })?;
+    let f = f.parse(args)?;
+    
     Ok(f)
 }
 
+// TODO change HashMap and Vec into wrapped structs
+pub struct FilterArguments<'a> {
+    pub positional: Box<Iterator<Item=Expression>>,
```

Comment:
Boxed iterator or just use `Vec`?  I had originally assumed the latter but have no concrete thoughts one way or the other.

Reply:
I chose the iterator because I think we actually only ever need to use `next()`. But I think there won't be a big difference regardless of what we choose so if you prefer I can turn it into a `Vec`.

Reply:
I honestly don't know if there is a reason to go one way or the other.  Just in my head when considering how to implement this, I was thinking of `Vec`. We ended up with different assumptions so I was curious if yours had any reasoning to it unlike mine.

Reply:
I think I started with your assumption (because of that outdated comment refering to a `Vec`) but for some reason I ended up changing it into an Iterator (this was before the holidays, so I may be forgetting the actual reason). I guess it was because of the use of `next`.

Reply:
Another good thing with the iterator is that it consumes each argument after calling `next`, which allows us to see whether or not all args were actually used (for `check_args_exhausted()`)